### DISCUSSION TITLE
Segmentation Fault on OSX

### DISCUSSION THREAD

**Discussion by chlangley** [tskit-dev/tsinfer#539] [Q&A] [answered]
Hello:
I followed the "Installation" to install tsinfer on my macOS laptop:
     'python3 -m pip install tsinfer --user chuck'  seemed to work and test with 'tsinfer --help' yielded the expected info.

When I tried the create a sample as in the tutorial:
'import cyvcf2
import tsinfer

def add_diploid_sites(vcf, samples):
    """
    Read the sites in the vcf and add them to the samples object, reordering the
    alleles to put the ancestral allele first, if it is available.
...
print(
    "Sample file created for {} samples ".format(samples.num_samples)
    + "({} individuals) ".format(samples.num_individuals)
    + "with {} variable sites.".format(samples.num_sites),
    flush=True,
)'
that seemed to work,
   "Sample file created for 20 samples (20 individuals) with 13192 variable sites."
 creating the two local files:
        P_dom_chr24_phased.samples
        P_dom_chr24_phased.vcf.gz.tbi

But when I run command:  'ts = tsinfer.infer(samples)'  python crashes with a segment fault.

"[1]    10164 segmentation fault  python3 infer_sample.py
/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d ' "

Sorry to bother this erudite discussion group with such mundane issues, but I am following the Documentation and we do want to attract users from far-afield who may make ignorant mistakes such as I am.

Thanks for any help.

Cheers,
Chuck

**Comment by hyanwong** [comment, marked as answer]
(also posted on Slack): If you are installing from source, there’s an issue on OSX with mixed compilers (https://github.com/tskit-dev/tsinfer/issues/376). You can get round it by installing the Conda C compilers in a separate env (`conda install -c conda-forge c-compiler`) and compiling there, or now that there are Conda packages, just

```
conda install tsinfer 
```

should do it. Let us know here if it works.

**Reply by chlangley** [reply]
Exploring all the suggestions I finally found that what you suggested here (conda install -c conda-forge c-compiler;  conda install tsinfer)  worked.  The was also an issue with cyvcf2 not yet being updated to be compatible with python 3.9.5.  But I was able to cp the files into my base env from a previously  pip-create directory.

**Reply by chlangley** [reply]
Thanks.  Chuck

**Reply by jeromekelleher** [reply]
Just as a follow-up here, there there shouldn't be any need to install the ``c-compiler`` if you're installing tsinfer from conda. The compilers are only needed for developers, really.

**Comment by chlangley** [comment]
Following that suggestion I tried this in a number of the suggested approaches without the c-compiler.  I could never get an installation in which the tsinfer infer command did not crash.  

Compiling from source seems to have done the trick.  Obviously, I don’t have a clue.

Cheers,
Chuck

**Reply by jeromekelleher** [reply]
Hmm, thanks Chuck, good to know.
